geohisto/intercommunalities.py

```diff
@@ -0,0 +1,287 @@
+    ('D ', 'DISTRICT '),
+    ('Di ', 'DISTRICT '),
+)
+
+TYPOS = {
+    'C C ': 'CC ',
+    'C c': 'CC ',
+    'C A ': 'CA ',
+    'C a': 'CA',
+    'C U ': 'CU ',
+    'C V ': 'CV ',
+    'S A N ': 'SAN ',
+    'C Agglo ': 'Communauté d\'Agglomération ',
+}
+
+COMMON_PREFIXES = (
+    'rural', 'rurale', 'ruraux', 'rurales',
+    'urbain', 'urbaine', 'urbains', 'urbaines'
+)
+
+PREPOSITIONS = ('du ', 'de ', 'de la ', 'd\'', 'des ')
+ALWAYS_LOWER = (
+    'du', 'de', 'des', 'sur', 'sous', 'en', 'la', 'le', 'les', 'aux', 'au', 'à'
+)
+WITH_APOSTROPHE = ('d\'', 'l\'')
+
+PRE_REPLACEMENTS = {
+    '': '\'',
+    '': 'œ',
+    '.': ' ',
+    ':': '',
+    '"': '',
+}
+
+POST_REPLACEMENTS = {
+    'Reg ': 'Région ',
+    'Region ': 'Région ',
+    'Cant ': 'Canton ',
+    'Breon': 'Bréon',
+    'Saone': 'Saône',
+    'Vallee': 'Vallée',
+    'D Or': 'd\'Or',
+    'Indre - Brenne': 'Indre-Brenne',
+    'Saône-Chalaronne': 'Saône Chalaronne',
+}
+
+
+def should_replace_kind(name, kind):
+    return name.startswith(tuple(
+        ' '.join((kind, prep)) for prep in PREPOSITIONS
+    )) or name.lower().startswith(tuple(
+        ' '.join((kind.lower(), prefix, prep)) for prep in PREPOSITIONS
+        for prefix in COMMON_PREFIXES
+    ))
+
+
+def fix_typos(name):
+    for typo, fix in TYPOS.items():
+        if typo in name:
+            name = name.replace(typo, fix)
+    return name
+
+
+def fix_word_casing(word):
+    if any(word.lower().startswith(prefix) for prefix in WITH_APOSTROPHE):
+        prefix, suffix = word.split('\'', 1)
+        return '\''.join((prefix.lower(), suffix.capitalize()))
+    elif word.lower() in ALWAYS_LOWER:
+        return word.lower()
+    elif word in NAME_RULES:
+        return word.upper()
+    else:
+        return word.capitalize()
+
+
+def fix_casing(name):
+    return ' '.join(fix_word_casing(word) for word in name.split(' '))
+
+
+def extract_name(line):
+    """Extract and normalize an intercommunality name"""
+    name = line['nom'].strip('\' ')
+    for char, replacement in PRE_REPLACEMENTS.items():
+        name = name.replace(char, replacement)
+    kind = line['nature']
+    name = fix_typos(name)
+    name = ' '.join(name.split())  # replace multiple spaces
+    if name.isupper():
+        name = fix_casing(name)
+    if line['nature'] == 'DISTRICT':
+        for prefix, replacement in DISTRICT_PREFIXES:
+            if name.startswith(prefix):
+                name = name.replace(prefix, replacement, 1)
+                break
+    if name.startswith(kind):
+        if should_replace_kind(name, kind):
+            name = name.replace(kind, NAME_RULES[kind])
+        else:
+            name = name.replace('{0} '.format(kind), '')
+    elif name.startswith(PREPOSITIONS):
+        name = ' '.join((NAME_RULES[kind], name))
+    elif name.lower().startswith(tuple(
+            ' '.join((kind.lower(), prefix, prep)) for prep in PREPOSITIONS
+            for prefix in COMMON_PREFIXES
+            )):
```

Comment:
Changed, but the files were already linted and linter was not complaining about this one.
Both are acceptable. Personnaly I prefer the version before the change, easier to read the 2 conditions.